vignettes/layer_templates.Rmd

```diff
@@ -0,0 +1,121 @@
+---
+title: "Layer Templates"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{layer_templates}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup, echo=FALSE}
+library(Tplyr)
+library(knitr)
+load('adsl.Rdata')
+```
+
+There are several scenarios where a layer template may be useful. Some tables, like demographics tables, may have many layers that will all essentially look the same. Categorical variables will have the same count layer settings, and continuous variables will have the same desc layer settings. A template allows a user to build those settings once per layer, then reference the template when the Tplyr table is actually built. Another scenario might be building a set of company layer templates that are built for standard tables to reduce the footprint of code across analyses. In either of these cases, the idea is the reduce the amount of redundant code necessary to create a table.
+
+Tplyr has already has a couple mechanisms to reduce redundant application of formats. For example, vignettes('tplyr_options') shows how the options `tplyr.count_layer_default_formats`, `tplyr.desc_layer_default_formats`, and `tplyr.shift_layer_default_formats` can be used to create default format string settings. Additionally, you can set formats table wide using `set_count_layer_formats()`, `set_desc_layer_formats()`, or `set_shift_layer_formats()`. But what these functions and options _don't_ allow you to do is pre-set and reuse the settings for an entire layer, so all of the additional potential layer modifying functions are ignored. This is where layer templates come in.
```

Comment:
I'm wondering if 'has a couple mechanisms' should be 'has a couple *of* mechanisms'. Could just be my writing style though.

Reply:
Also, may want back ticks around "vignettes('tplyr_options')"

Reply:
Updated